internal/locate/region_cache.go

```diff
@@ -1187,6 +1187,46 @@ func (b *Bucket) Contains(key []byte) bool {
 	return contains(b.StartKey, b.EndKey, key)
 }
 
+// clampBucketToRegion ensures bucket boundaries don't exceed region boundaries.
+// This is a defensive measure against stale bucket metadata that can have keys
+// outside the current region boundary.
+func (l *KeyLocation) clampBucketToRegion(bucket *Bucket) *Bucket {
+	startKey := bucket.StartKey
+	endKey := bucket.EndKey
+
+	// Clamp start: max(bucket.StartKey, region.StartKey)
+	if len(l.StartKey) > 0 && (len(startKey) == 0 || bytes.Compare(startKey, l.StartKey) < 0) {
```

Comment:
Isn't the case completely included in `bytes.Compare(startKey, l.StartKey) < 0`?